README.md

```diff
@@ -134,13 +134,15 @@ It is possible to attach a remote debugger to the Jenkins instance started by `g
 environment variable must be used to configure the JVM debug options.
 
     $ export GRADLE_OPTS=-agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=5005
-    $ ./gradlew server
+    $ ./gradlew --no-daemon server
+
+The --no-daemon options prevents gradle from spawning another process to run the build (and Jenkins)
 
 The `server` task enables several debug options: `stapler.trace`, `stapler.jelly.noCache` and `debug.YUI`. This
 increases the page load time. All option can be changed and new options can be added by passing them as system
 properties to the Gradle command line.
 
-    $ ./gradlew -Dstapler.jelly.noCache=false server
+    $ ./gradlew --no-daemon -Dstapler.jelly.noCache=false server
```

Comment:
Why do we need the `no-daemon` option here?

Reply:
Since gradle 3.0, gradle no longer use its main process to do the build. Instead, it spawns long lived daemons. 
https://docs.gradle.org/current/userguide/gradle_daemon.html
As a result when attaching the debugger, it gets attached to the main gradle process, but not to the daemon that actually loads and launches Jenkins, thus preventing debugging. 
—no-daemon ensures that the build takes place in a single process. 
It is the only way I found to attach the Jenkins instance in IntelliJ anyway. 
Does it work for you without the flag?

Reply:
Sorry, I should have been more clear: Why do we need `--no-daemon` if we only want to invoke the server with some special system properties.

Now that I think of it: did you try using `-Dorg.gradle.debug=true`?

I think that the problem using `GRADLE_OPTS` is that they are passed to the wrapper, but not the daemon. In order to set the debug parameters for the Gradle daemon, you need to use the [Gradle property](https://docs.gradle.org/current/userguide/build_environment.html#sec:gradle_configuration_properties) `org.gradle.jvmargs`. So using 
```
./gradlew server -Dorg.gradle.jvmargs=-agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=5005
```

should also do the trick.
Using `-Dorg.gradle.debug=true` seems to be simpler, though.

Reply:
Thanks for the comments. I tried `-Dorg.gradle.debug=true` and it is indeed working as intended.